extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,621 @@
+
+		$mock = $this->getMockBuilder( 'Wikia\PortableInfobox\Helpers\PortableInfoboxRenderServiceHelper' )
+			->setMethods( [ 'isValidHeroDataItem', 'validateType', 'isMercury',
+				'isMercuryExperimentalMarkupEnabled', 'createHorizontalGroupData', 'extendImageData' ] )
+			->getMock();
+		$mock->expects( $this->any() )
+			->method( 'isValidHeroDataItem' )
+			->will( $this->returnValue( $isValidHeroDataItem ) );
+		$mock->expects( $this->any() )
+			->method( 'validateType' )
+			->will( $this->returnValue( true ) );
+		$mock->expects( $this->any() )
+			->method( 'isMercury' )
+			->will( $this->returnValue( $isMercury ) );
+		$mock->expects( $this->any() )
+			->method( 'createHorizontalGroupData' )
+			->will( $this->returnValue( $createHorizontalGroupData ) );
+		$mock->expects( $this->any() )
+			->method( 'extendImageData' )
+			->will( $this->returnValue( $extendImageData ) );
+
+		$this->mockClass( 'Wikia\PortableInfobox\Helpers\PortableInfoboxRenderServiceHelper', $mock );
+	}
+
+	/**
+	 * @param $html
+	 * @return string
+	 */
+	private function normalizeHTML( $html ) {
+		$DOM = new DOMDocument( '1.0' );
+		$DOM->formatOutput = true;
+		$DOM->preserveWhiteSpace = false;
+		$DOM->loadXML( $html );
+
+		return $DOM->saveXML();
+	}
+
+	/**
+	 * @param $input
+	 * @param $expectedOutput
+	 * @param $description
+	 * @param $mockParams
+	 * @dataProvider testRenderInfoboxDataProvider
+	 */
+	public function testRenderInfobox( $input, $expectedOutput, $description, $mockParams ) {
+		$this->mockInfoboxRenderServiceHelper( $mockParams );
+
+		$infoboxRenderService = new PortableInfoboxMobileRenderService();
+		$actualOutput = $infoboxRenderService->renderInfobox( $input, '', '' );
+		$expectedHtml = $this->normalizeHTML( $expectedOutput );
+		$actualHtml = $this->normalizeHTML( $actualOutput );
+
+		$this->assertEquals( $expectedHtml, $actualHtml, $description );
+	}
+
+	public function testRenderInfoboxDataProvider() {
+		return [
+			[
+				'input' => [ ],
+				'output' => '',
+				'description' => 'Empty data should yield no infobox markup'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<h2 class="pi-item pi-item-spacing pi-title">Test Title</h2>
+							</aside>',
+				'description' => 'Only title'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'navigation',
+						'data' => [
+							'value' => 'navigation value',
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<nav class="pi-navigation pi-item-spacing pi-secondary-background pi-secondary-font">navigation value</nav>
+							</aside>',
+				'description' => 'navigation only'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'data',
+						'data' => [
+							'label' => 'test label',
+							'value' => 'test value'
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "test value" 14 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)